huff_utils/src/sol_interface.rs

```diff
@@ -0,0 +1,93 @@
+                defs.push(format!(
+                    "{}event {}({});",
+                    "\t",
+                    f.name,
+                    f.inputs
+                        .iter()
+                        .map(|i| {
+                            format!(
+                                "{}{}",
+                                i.kind,
+                                if i.indexed {
+                                    String::from(" indexed")
+                                } else {
+                                    String::default()
+                                }
+                            )
+                        })
+                        .collect::<Vec<_>>()
+                        .join(", "),
+                ));
+            });
+            a.functions.iter().for_each(|(_, f)| {
+                defs.push(format!(
+                    "{}function {}({}) external{}{};",
+                    "\t",
+                    f.name,
+                    f.inputs
+                        .iter()
+                        .map(|i| format!(
+                            "{}{}",
+                            i.kind,
+                            if i.kind.is_memory_type() { " memory" } else { "" }
+                        ))
+                        .collect::<Vec<_>>()
+                        .join(", "),
+                    f.state_mutability.interface_mutability(),
+                    if f.outputs.is_empty() {
+                        String::default()
+                    } else {
+                        format!(
+                            " returns ({})",
+                            f.outputs
+                                .iter()
+                                .map(|o| format!(
+                                    "{}{}",
+                                    o.kind,
+                                    if o.kind.is_memory_type() { " memory" } else { "" }
+                                ))
+                                .collect::<Vec<_>>()
+                                .join(", ")
+                        )
+                    },
+                ));
+            });
+
+            let interface_name =
+                artifact.file.path.split('/').last().unwrap().split('.').next().unwrap();
+            interfaces.push((
+                interface_name,
+                format!("interface I{} {{\n{}\n}}", interface_name, defs.join("\n"),),
+            ));
+        }
+    }
+
+    interfaces
+}
+
+/// Export generated solidity interfaces to a file.
+///
+/// @param interfaces The vector of generated interfaces.
+/// @return Unit type if success, error if failure.
+pub fn export_interfaces(interfaces: &Vec<(&str, String)>) -> Result<(), std::io::Error> {
+    for (name, interface) in interfaces {
+        let path_str = format!("./I{}.sol", name);
+        let file_path = Path::new(&path_str);
+        fs::write(file_path, interface)?;
+    }
+    Ok(())
+}
```

Comment:
Only thing: I think we'll want to save the interface file in the same location as the source. Maybe we can make this function accept a prefix file which is the path of the source (less the source file) and we set the `path_str` to

```
let path_str = format!("{}I{}.sol", prefix, name);
```

Reply:
Should be added :smile: